modules/effects/src/main/scala/dev/profunktor/redis4cats/algebra/keys.scala

```diff
@@ -46,6 +46,7 @@ trait KeyCommands[F[_], K] {
   @deprecated("In favor of scan(cursor: KeyScanCursor[K], scanArgs: ScanArgs)", since = "0.10.4")
   def scan(cursor: Long, scanArgs: ScanArgs): F[KeyScanCursor[K]]
   def scan(previous: KeyScanCursor[K], scanArgs: ScanArgs): F[KeyScanCursor[K]]
+  def typeOf(key: K): F[Option[RedisType]]
```

Comment:
Should it return an `Option[RedisType]` or if the type is unknown fail with a custom exception?

Reply:
I dont see how there can be an unknown type , as the types are not user defined and are enumerated by Redis.
The reason for the Option is because Redis returns a `none` string if key doesnt exist .

Reply:
Ah that makes sense :)